### PR TITLE
add progress bar - issue #85

### PR DESCRIPTION
**Pull request by gilad1bl**
I figured it only make sense to add the progress bar if there is more than iteration of downloading (that is, more than 25 results if the default setting is not changed).
I'm not sure this is the cleanest way to implement this -  I used the function that Michael suggested and added it directly to the search class. Right now it is implemented only in Scopus Search but, if the proposed solution is agreed upon, it is very easy to duplicate to the other searching methods.

### CODE REVIEW

**Comment by Michael-E-Rose** [comment]
Thanks for this, @gilad1bl ! Yes, I would much prefer to have this in all Search classes.

Could you please move the function to `/scopus/utils/get_content.py` and import from there?  And then rename the corresponding parameter `verbose` instead of `print_PB`?  Also, a one-line explanation in the docstring is necessary.

**Comment by gilad1bl** [comment]
thanks @Michael-E-Rose, no problem- ill do it within the next few days

**Comment by gilad1bl** [comment]
Hi @Michael-E-Rose , I moved the function to /scopus/utils/get_content.py, change parameter name to verbose, added it to all three Search classes, and added a short explanation.

**Comment by gilad1bl** [comment]
Oh, and it's all updated to the latest version of pybliometrics, so some of the changes seen are changes made in the package since this PR was created (before it was named pybliometrics)

File: scopus/utils/get_content.py [hunk 1]
```diff
@@ -167,3 +168,13 @@ def get_folder(api, view):
...
+
+
+def print_progress(iteration, total, length=50):
+    """Print terminal progress bar."""
+    percent = 100 * (iteration / float(total))
+    filled_len = int(length * iteration // total)
+    bar = '█' * filled_len + '-' * (length - filled_len)
+    print('\rProgress: |{}| {:.2f}% Complete'.format(bar, percent), end='\r')
+    if iteration == total:
+        print()
```
Comment (Michael-E-Rose) [inline scopus/utils/get_content.py]: Newline at the end is missing

File: scopus/affiliation_search.py [hunk 2]
```diff
@@ -46,6 +46,9 @@ def __init__(self, query, refresh=False, download=True, count=200):
         download : bool (optional, default=True)
             Whether to download results (if they have not been cached).
 
+        verbose : bool (optional, default=False)
+            Whether to print a downloading progress bar to terminal. requiers download=True.
```
Comment (Michael-E-Rose) [inline scopus/affiliation_search.py]: Capital letter after full stop. Maybe it's better to write "Has no effect for download=False".  It sounds less harsh to me.

File: scopus/author_search.py [hunk 3]
```diff
@@ -58,6 +58,9 @@ def __init__(self, query, refresh=False, count=200, download=True):
         download : bool (optional, default=True)
             Whether to download results (if they have not been cached).
 
+        verbose : bool (optional, default=False)
+            Whether to print a downloading progress bar to terminal. requiers download=True.
```
Comment (Michael-E-Rose) [inline scopus/author_search.py]: Same as above.

File: scopus/scopus_search.py [hunk 4]
```diff
@@ -113,14 +113,12 @@ def __init__(self, query, refresh=False, subscriber=True,
...
-        cursor : bool (optional, default=True)
-            Whether to use Scopus's cursor navigation to obtain results.
-            Using the cursor allows to download an unlimited results set.
-            Non-subscribers should set this to False.
-
         download : bool (optional, default=True)
             Whether to download results (if they have not been cached).
 
+        verbose : bool (optional, default=False)
+            Whether to print a downloading progress bar to terminal. requiers download=True.
```
Comment (Michael-E-Rose) [inline scopus/scopus_search.py]: Same as above

**Comment by Michael-E-Rose** [comment]
Looks like a very decent job!  Just those few formatting things and I'll merge the PR.

**Comment by gilad1bl** [comment]
Thank you @Michael-E-Rose for your comments! I edited it just as you suggested, it sounds great.

**Comment by Michael-E-Rose** [comment]
All that's left is the newline in `scopus/utils/get_content.py`.

Do you have an idea what the conflicts might be?   On my end it says they are too complex to resolve in the web.

**Comment by gilad1bl** [comment]
Hey, I added a newline in `scopus/utils/get_content.py`. Currently I really have no idea what the conflicts could be, as I mentioned, I updated the fork 3 days ago to the latest version. I will gladly try to figure it out, but it would take me about a week to get to it (I guess the timeline is not very tight since this is not really a core function).

**Comment by katrinleinweber** [comment]
Hello :-) I think the conflicts started with https://github.com/pybliometrics-dev/pybliometrics/commit/2983b48f8dc43931d8489233363850742fd0b2a8.

I tried `git checkout gilad1bl/master && git rebase upstream/master` and [they didn't seem too bad to resolve](https://github.com/pybliometrics-dev/pybliometrics/compare/master...katrinleinweber:pr-103?expand=1#diff-7ea7a880ab5840cec7d35d0cc2ed1fcb).

**Comment by gilad1bl** [comment]
thanks @katrinleinweber !! you are probably right, but still I don't understand why it would raise a conflict? The current PR is inline with the master branch...what am I missing?
Anyway, I'll send your fixed PR.

**Comment by katrinleinweber** [comment]
What exactly do you mean with "inline with the master branch"? The master branches of your fork and the main repo can diverge and have in this case, see https://github.com/pybliometrics-dev/pybliometrics/network.

To avoid this in the future, the [approach of creating yet another branch in your fork](https://opensource.guide/how-to-contribute/#opening-a-pull-request) helps a bit, as does frequent `git merge upstream/master` (or `rebase`), as does quickly completing a PR. Nothing except "hands up!" _really_ prevents conflicts ;-)

**Comment by Michael-E-Rose** [comment]
Katrin is exactly right.

Also, I can't merge the PR unless the conflicts have been resolved on your side.

Maybe at this stage it's easier to fork the repo again, create a branch, add your changes, and the submit the PR again.

**Pull request closed without merging**